### PR TITLE
Remove all build errors on Android Studio 3.2.0

### PR DESCRIPTION
**Pull request by hevrard**
Update Gradle version and fix build files in order to get rid of all
warnings and errors when building through the latest Android Studio
3.2.0. Also remove some obsolete entries of the AndroidManifest,
which are now set by Gradle.

### CODE REVIEW

**Comment by cnorthrop** [comment]
We're looking into the lint error triggered by CI.

**Comment by ggfan** [comment]
That android.permission.SET_DEBUG_APP most likely was copied in from the original NativeActivity sample, let's delete it: it does not make any difference for application.

File: API-Samples/android/project_template/build.gradle [hunk 1]
```diff
@@ -32,13 +32,13 @@ if(!ndkDir || ndkDir.empty) {
 def stlType = 'c++_static'
 
 android {
-    compileSdkVersion  24
-    buildToolsVersion '25.0.0'
+    compileSdkVersion  26
```
Comment (ggfan) [inline API-Samples/android/project_template/build.gradle]: change this to 28, or leave it to 26 is fine.

File: API-Samples/android/project_template/build.gradle [hunk 2]
```diff
@@ -32,13 +32,13 @@ if(!ndkDir || ndkDir.empty) {
...
+    buildToolsVersion '28.0.2'
```
Comment (ggfan) [inline API-Samples/android/project_template/build.gradle]: delete this line: let Android studio find the matching tools version for "compileSdkVersion"

File: API-Samples/android/project_template/build.gradle [hunk 3]
```diff
@@ -32,13 +32,13 @@ if(!ndkDir || ndkDir.empty) {
...
 
     defaultConfig {
         applicationId '@PACKAGE_NAME@'
         minSdkVersion    22
```
Comment (ggfan) [inline API-Samples/android/project_template/build.gradle]: theoretically, we could change this minSdkVersion to 24, as formally it is 24 that begins supporting Vulkan.

**Comment by hevrard** [comment]
Thanks for the guidance!
Does this look good to be merged, and if so, do you know who's able to also update the googlesamples/vulkan-basic-samples repo accordingly?

**Comment by ggfan** [comment]
I will update the downstream repo after this is merged.

**Comment by ggfan** [comment]
@cnorthrop may you kindly merge this one? Thanks!

**Comment by cnorthrop** [comment]
Will do, thanks Gerry!

**Pull request merged**